### PR TITLE
JAMES-2402 ToRepositoryTest was unstable

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

File: server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/ToRepositoryTest.java [hunk 1]
```diff
@@ -153,7 +158,7 @@ public void userShouldBeAbleToAccessReprocessedMail() throws Exception {
             .login(RECIPIENT, PASSWORD)
             .select(IMAPMessageReader.INBOX)
             .awaitMessage(awaitAtMostOneMinute)
```
Comment (mbaechler) [inline server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/ToRepositoryTest.java]: this line looks useless now

File: server/testing/src/main/java/org/apache/james/utils/IMAPMessageReader.java [hunk 2]
```diff
@@ -82,6 +77,27 @@ public IMAPMessageReader awaitMessage(ConditionFactory conditionFactory) throws
         return this;
     }
 
+    public IMAPMessageReader awaitMessageCount(ConditionFactory conditionFactory, int messageCount) {
+        conditionFactory.until(() -> {
+            try {
+                imapClient.fetch("1:*", "ALL");
+                return  countFetchedEntries() == messageCount;
```
Comment (rouazana) [inline server/testing/src/main/java/org/apache/james/utils/IMAPMessageReader.java]: 2 spaces

Comment (chibenwa) [reply server/testing/src/main/java/org/apache/james/utils/IMAPMessageReader.java]: Ha...

Comment (chibenwa) [reply server/testing/src/main/java/org/apache/james/utils/IMAPMessageReader.java]: Funny that it's not caught by checkstyle...

Comment (rouazana) [reply server/testing/src/main/java/org/apache/james/utils/IMAPMessageReader.java]: I tried to find why, but failed. But it seems that checkstyle is ignoring some tests files.

**Comment by mbaechler** [comment]
merged

**Pull request closed without merging**